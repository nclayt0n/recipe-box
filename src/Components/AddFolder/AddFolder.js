import React from 'react'
import {withRouter} from 'react-router-dom'
import Context from '../../Context'
import config from '../../config'
import hpStyles from '../HomePage/HomePageStyles'
import addFStyles from './AddFolderStyles'
import Header from '../Header/Header'
import Nav from '../Nav/Nav'
import ValidationError from '../../Validation/ValidationError'
import TokenService from '../../services/token-service'

class AddFolder extends React.Component{
    static contextType=Context;
    constructor(){
        super()
        this.state={
            error:''
        }
    }
    handleSubmit=(e)=>{
        e.preventDefault();
        const folderName=(e.target.folderName.value)
        if(folderName.length<3){
            this.setState({error:'Name Must contain at least 3 characters'})
        }
        if(folderName.length>=3){

            const url=`${config.API_ENDPOINT}/folders`;
            const options={
                method:'POST',
                headers:{
                    'content-type':'application/json',
                    'Authorization': `Bearer ${TokenService.getAuthToken()}`,
                },
                body: JSON.stringify({'name':folderName,'user_id':this.context.user_id})
            };

            fetch(url,options)
            .then(response => {
                if (response.ok) {
                return response.json();
                }
                throw new Error(response.statusText);
            })
            .then(responseJson =>this.context.addFolder(responseJson))
            .catch(error =>{
                console.error(error)
            })
            if(this.props.location.pathname==='/home-page'){
                this.props.history.push('/home-page')
            }else{
                this.props.history.push('/folder-list')
            }  
        }
    }
    render(){ 
        let style;
         if(this.props.location.pathname===`/home-page`){
            style=hpStyles.addFolder
        }if(this.props.location.pathname==='/add-folder'){
            style=addFStyles
        }
        return(
        <>
        {(this.props.location.pathname===`/home-page`)?'':<Header/>}
        {(this.props.location.pathname===`/home-page`)?'':<Nav userId={this.context.recipes[0].user.id}/>}
        <div className='addFolder' style={style.addFolderDivStyle} >
            <h3>ADD FOLDER</h3>
            <form onSubmit={(e)=>this.handleSubmit(e)}>
                <fieldset>
                    
                    <label htmlFor ='folderName'>Folder Name:
                    {(this.props.location.pathname===`/home-page`)?null:<br/>} 
                    <input type='text' name='folderName' style={style.addFolderInput}/></label> 
                    {(this.props.location.pathname===`/home-page`)?null:<br/>} 
                     <ValidationError Namemessage={this.state.error}/>
                    <button type='submit' style={{width:'100px', fontSize: '1vw',}}>Submit
                    </button>
                </fieldset>
            </form> 
            {(this.props.location.pathname===`/home-page`)?null:<button onClick={()=>this.props.history.push('/home-page')} style={{width:'100px',margin:'auto', fontSize: '1vw'}}>Cancel</button>} 
        </div>
        </>
        )
    }
}
export default withRouter(AddFolder)