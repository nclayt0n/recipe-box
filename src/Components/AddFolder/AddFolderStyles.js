import React from 'react'
import backgroundImage from '../../images/purplelavleaves.jpg'
const addFStyles = {
    addFolderDivStyle: {
        backgroundImage: `url(${backgroundImage})`,
        backgroundRepeat: 'no-repeat',
        backgroundSize: '100%',
        backgroundColor: 'white',
        backgroundPosition: 'top',
        border: '2px solid var(--purple)',
        textAlign: 'center',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        width: '70%',
        margin: '50px auto',
        fontSize: '2vw',
        padding: '50px',
    },
    addFolderLabel: {
        fontSize: '3vw',
        color: 'var(--purple)'
    },
    addFolderInput: {
        width: '70%',
        fontSize: '2.5vw',
        margin: '5px auto',

    },
    addFolderButtons: {
        width: '100px',
        backgroundColor: 'var(--purple)',
        color: 'white',
        margin: 'auto',
        fontSize: '3vw',
    },
    h3: {
        color: 'var(--purple)',
        textAlign: 'center',
        textShadow: '1px 1px #8a64a3',
        fontSize: '3.25vw',
    },
    addFolderFieldset: {
        border: 'none'
    },
    tablet: {
        addFolderDivStyle: {
            backgroundImage: `url(${backgroundImage})`,
            backgroundRepeat: 'no-repeat',
            backgroundSize: '100%',
            backgroundColor: 'white',
            backgroundPosition: 'top',
            border: '2px solid var(--purple)',
            textAlign: 'center',
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'center',
            width: '50%',
            margin: '50px auto ',
            padding: '40px',
        },
        addFolderLabel: {
            fontSize: '2vw',
            color: 'var(--purple)'
        },
        addFolderInput: {
            width: '50%',
            fontSize: '2vw',
            margin: '5px auto',

        },
        addFolderButtons: {
            width: '100px',
            backgroundColor: 'var(--purple)',
            color: 'white',
            margin: 'auto',
            fontSize: '2vw',
        },
        h3: {
            color: 'var(--purple)',
            textAlign: 'center',
            textShadow: '1px 1px #8a64a3',
            fontSize: '2.75vw',
        },

    },
    laptop: {
        addFolderDivStyle: {
            backgroundImage: `url(${backgroundImage})`,
            backgroundRepeat: 'no-repeat',
            backgroundSize: '100%',
            backgroundColor: 'white',
            backgroundPosition: 'top',
            border: '2px solid var(--purple)',
            textAlign: 'center',
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'center',
            width: '50%',
            margin: '50px auto ',
            padding: '40px',
        },
        addFolderLabel: {
            fontSize: '1.25vw',
            color: 'var(--purple)'
        },
        addFolderInput: {
            width: '40%',
            fontSize: '1.25vw',
            margin: '5px auto',

        },
        addFolderButtons: {
            width: '100px',
            backgroundColor: 'var(--purple)',
            color: 'white',
            margin: 'auto',
            fontSize: '1.25vw',
        },
        h3: {
            color: 'var(--purple)',
            textAlign: 'center',
            textShadow: '1px 1px #8a64a3',
            fontSize: '1.75vw',
        },
    }

}


export default addFStyles