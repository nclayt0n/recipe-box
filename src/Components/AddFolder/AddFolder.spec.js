import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import AddFolder from './AddFolder';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import addFStyles from './AddFolderStyles';

        
it('renders without crashing', () => {
    let style=addFStyles;
 const wrapper=shallow(<AddFolder style={style.addFDivStyle}/>)
 expect(toJson(wrapper))
 .toMatchSnapshot();

});