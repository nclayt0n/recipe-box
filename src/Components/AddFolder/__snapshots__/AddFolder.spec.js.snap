// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <withRouter(AddFolder)
    location={
      Object {
        "pathname": "/add-folder",
      }
    }
    style={
      Object {
        "backgroundColor": "white",
        "backgroundImage": "url(purplelavleaves.jpg)",
        "backgroundPosition": "top",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "100%",
        "border": "2px solid var(--purple)",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "margin": "50px auto",
        "padding": "30px",
        "textAlign": "center",
        "width": "300px",
      }
    }
  />
</Router>
`;
